test/e2e/case19_ns_selector_test.go

```diff
@@ -0,0 +1,124 @@
+				"[configmap-selector-e2e] in namespace case19-5-e2e missing",
+		},
+		{
+			"LabelSelector",
+			"{\"matchExpressions\":[{\"key\":\"name\",\"operator\":\"Exists\"}]}",
+			"configmaps not found: [configmap-selector-e2e] in namespace case19-2-e2e missing; " +
+				"[configmap-selector-e2e] in namespace case19-3-e2e missing; " +
+				"[configmap-selector-e2e] in namespace case19-4-e2e missing; " +
+				"[configmap-selector-e2e] in namespace case19-5-e2e missing",
+		},
+	}
+
+	It("creates prerequisite objects", func() {
+		utils.Kubectl("apply", "-f", case19PrereqYaml)
+		// Delete the last namespace so we can use it to test whether
+		// adding a namespace works as the final test
+		utils.Kubectl("delete", "namespace", "case19-6-e2e")
+		utils.Kubectl("apply", "-f", case19PolicyYaml, "-n", testNamespace)
+		plc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+			case19PolicyName, testNamespace, true, defaultTimeoutSeconds)
+		Expect(plc).NotTo(BeNil())
+	})
+
+	It("should properly handle the namespaceSelector", func() {
+		for _, patch := range patches {
+			By("patching compliant policy " + case19PolicyName + " on the managed cluster")
+			utils.Kubectl("patch", "--namespace=managed", "configurationpolicy", case19PolicyName, "--type=json",
+				"--patch="+case19PatchPrefix+resetPatch+case19PatchSuffix,
+			)
+			Eventually(func() interface{} {
+				managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+					case19PolicyName, testNamespace, true, defaultTimeoutSeconds)
+
+				return utils.GetComplianceState(managedPlc)
+			}, defaultTimeoutSeconds, 1).Should(Equal("Compliant"))
+			By("patching with " + patch.description)
+			utils.Kubectl("patch", "--namespace=managed", "configurationpolicy", case19PolicyName, "--type=json",
+				"--patch="+case19PatchPrefix+patch.patch+case19PatchSuffix,
+			)
+			Eventually(func() interface{} {
+				managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+					case19PolicyName, testNamespace, true, defaultTimeoutSeconds)
+
+				return utils.GetComplianceState(managedPlc)
+			}, defaultTimeoutSeconds, 1).Should(Equal("NonCompliant"))
+			Eventually(func() interface{} {
+				managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+					case19PolicyName, testNamespace, true, defaultTimeoutSeconds)
+
+				return utils.GetStatusMessage(managedPlc)
+			}, defaultTimeoutSeconds, 1).Should(Equal(patch.message))
+		}
+	})
+
+	It("should handle when a matching labeled namespace is added", func() {
+		utils.Kubectl("apply", "-f", case19PrereqYaml)
+		Eventually(func() interface{} {
+			managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+				case19PolicyName, testNamespace, true, defaultTimeoutSeconds)
+
+			return utils.GetStatusMessage(managedPlc)
+		}, defaultTimeoutSeconds, 1).Should(ContainSubstring(
+			"[configmap-selector-e2e] in namespace case19-6-e2e missing"))
+	})
+
+	It("Cleans up", func() {
```

Comment:
If this was implemented as an `AfterAll`, then it would always cleanup after itself. The other tests here are ordered anyway, so it would be good to add. https://onsi.github.io/ginkgo/#setup-in-ordered-containers-beforeall-and-afterall